### PR TITLE
PP 50 upload photos video - Teacher Profile

### PR DESCRIPTION
**Pull request by vrudnitskaya**
### Description
The function of uploading images and videos to the teacher's portfolio has been implemented.
### Screenshots
photo-video-portfolio1

The input for uploading files is shown after clicking the add button. There is also information about file types and maximum size.

show-file-input

photo-video-portfolio2

The users have an opportunity to upload only one welcome video. Until they delete the current welcome video, they can't add a new one. 
photo-video-portfolio3

It is also possible to upload several files at once for the portfolio.
photo-video-portfolio4

### CODE REVIEW

File: src/components/portfolio/FormUploadPortfolio.jsx
Comment (mariyador) [inline src/components/portfolio/FormUploadPortfolio.jsx:1]: Valya!

That’s great! I can add a welcome video and files to my portfolio.

However, when I’ve already uploaded a file and then try to add more but forget to select a new file, and push upload, the request still goes through and reuses the previously uploaded file(s).

**Review by mariyador — CHANGES_REQUESTED** [review]

File: src/components/portfolio/FormUploadPortfolio.jsx
Comment (vrudnitskaya) [reply src/components/portfolio/FormUploadPortfolio.jsx:3]: fixed

**Review by mariyador — APPROVED** [review]
🥰

File: src/components/pages/dashboard-pages/HomeTeacher.jsx
Comment (AnnaPestova1) [inline src/components/pages/dashboard-pages/HomeTeacher.jsx:66]: Great to vreate additional components for each functionality

File: src/components/portfolio/FormUploadPortfolio.jsx
Comment (AnnaPestova1) [inline src/components/portfolio/FormUploadPortfolio.jsx:6]: Do we really need this `useEffect`? without dependencies it works only once during loading the component, but we already have `useState` with default `isFileSelected` value === false

File: src/util/PortfolioUploadMediaService.js
Comment (AnnaPestova1) [inline src/util/PortfolioUploadMediaService.js:46]: do not forget to delete unused console.logs

**Review by AnnaPestova1 — CHANGES_REQUESTED** [review]
Hi Valya! 
Great realization! I really like how you are coding. I have a question about a useEffect (see comments) and, by the way, I was able to download a .mov video

**Comment by vrudnitskaya** [comment]
> Hi Valya! Great realization! I really like how you are coding. I have a question about a useEffect (see comments) and, by the way, I was able to download a .mov video

If there were no errors from the backend and the video was successfully uploaded, then I think everything is fine. I will add this format as available for upload.

File: src/components/portfolio/FormUploadPortfolio.jsx
Comment (JoHeyJo) [inline src/components/portfolio/FormUploadPortfolio.jsx:24]: You might want to adjust the margins/padding so that hovering over "Upload" or "Cancel" it doesn't shift the contents of the container.

File: src/components/portfolio/VideoPortfolio.jsx
Comment (JoHeyJo) [inline src/components/portfolio/VideoPortfolio.jsx:36]: "delete video"

File: src/components/portfolio/FormUploadPortfolio.jsx
Comment (JoHeyJo) [inline src/components/portfolio/FormUploadPortfolio.jsx:27]: This is cut off on smaller displays.

File: src/components/portfolio/WelcomeVideo.jsx
Comment (JoHeyJo) [inline src/components/portfolio/WelcomeVideo.jsx:35]: Also "delete video" you already have alt describing that nested inside this a delete image with "delete icon"

File: src/util/PortfolioUploadMediaService.js
Comment (JoHeyJo) [inline src/util/PortfolioUploadMediaService.js:14]: Should we reset the state after successful upload or are we sure that there won't be any residual files that will be uploaded again?

File: src/util/PortfolioUploadMediaService.js
Comment (JoHeyJo) [inline src/util/PortfolioUploadMediaService.js:1]: Nice use of a custom hook to handle this logic. 
Stretch goal: This is a pretty big file. Can we refactor some of the code to remove some repetition? There are a lot of similarities with the request logic and error handling logic.

**Review by JoHeyJo — CHANGES_REQUESTED** [review]
This is incredible! Well done. There are a few things to look over before approval.

**Review by AnnaPestova1 — COMMENTED** [review]
Thank you Valya! Great work. Approved, but you need to handle Joannes's comments and receive his approval before merge. So I leave this message as comment, to wait the other approval from  Joannes

File: src/util/PortfolioUploadMediaService.js
Comment (vrudnitskaya) [reply src/util/PortfolioUploadMediaService.js:1]: I tried really hard to reduce repetition where I could by combining uploading/deleting videos and photos for a portfolio. I'll try to come back to this a little later, when I finish another feature.

**Comment by vrudnitskaya** [comment]
> This is incredible! Well done. There are a few things to look over before approval.

Thank you!

File: src/util/PortfolioUploadMediaService.js
Comment (vrudnitskaya) [reply src/util/PortfolioUploadMediaService.js:14]: Ok. I'll check this state.

File: src/components/portfolio/ImagePortfolio.jsx
Comment (JoHeyJo) [inline src/components/portfolio/ImagePortfolio.jsx:52]: Stretch goal: I noticed when you click on the delete there is no indication that a request has been made. This is fine for now but it would be nice to show the user that the image is being deleted or to at least disable the delete button.

File: src/util/PortfolioUploadMediaService.js
Comment (vrudnitskaya) [reply src/util/PortfolioUploadMediaService.js:14]: I checked, and indeed the file data remains in the state. But due to the fact that the form is updated and the upload button is blocked until a new file is selected, the old file is not sent. I added resetting this state anyway. Thank you for your comment.

File: src/components/portfolio/FormUploadPortfolio.jsx
Comment (vrudnitskaya) [reply src/components/portfolio/FormUploadPortfolio.jsx:24]: fixed it by changing the border settings

File: src/components/portfolio/FormUploadPortfolio.jsx
Comment (vrudnitskaya) [reply src/components/portfolio/FormUploadPortfolio.jsx:27]: I've added the min width to the button

File: src/components/portfolio/ImagePortfolio.jsx
Comment (vrudnitskaya) [reply src/components/portfolio/ImagePortfolio.jsx:52]: Thanks for the comment, added the loading state for delete functions.

File: src/util/PortfolioUploadMediaService.js
Comment (JoHeyJo) [reply src/util/PortfolioUploadMediaService.js:14]: Good catch! Better to be safe then sorry.

File: src/util/PortfolioUploadMediaService.js
Comment (JoHeyJo) [reply src/util/PortfolioUploadMediaService.js:1]: Don't worry about this took much right now. It's crunch time and your application is working as it should. Take this as more of a reminder when you are working on your next codebase.

File: src/components/portfolio/ImagePortfolio.jsx
Comment (JoHeyJo) [reply src/components/portfolio/ImagePortfolio.jsx:52]: Are you able to instead disable the delete icon while the delete request is being made. This will indicate that the request has been made and will prevent spamming of the button. This way the functionality of deleting multiple images at once isn't obscured by the loading circle. Ideally you want each image rendering the loading circle within its container when the delete request is being made. **Do not do this now.** This might be a little involved, just a design tip to keep in mind for the future.

**Review by JoHeyJo — APPROVED** [review]
This is a fantastic feature, great job! If you choose to go with the loading spinner or just disabling delete it's up to you. The idea is that you just want some kind of indication to the user that the request has been made. There are many ways to do this but for now let's just offer something simple that gets the point across.

**Review by AnnaPestova1 — APPROVED** [review]
Great job Valya! As always

**Pull request merged**
